### PR TITLE
Add RISC-V support to thor

### PR DESCRIPTION
**Pull request by avdgrinten**
This PR makes thor compile, even though it (obviously) does not work yet.

We also use clang-format for the newly added files in `kernel/thor/arch/riscv` and add the appropriate configuration files for that purpose.

Original code by @ElectrodeYT, rebased by @no92 and rebased again by me.

Supersedes #534.

### CODE REVIEW

File: kernel/thor/arch/riscv/link.x [hunk 1]
```diff
@@ -0,0 +1,87 @@
+ENTRY(thorRtEntry)
+
+SECTIONS {
+	/* thor starts a bit higher than normal, to make space for eir on EFI systems. */
+	. = 0xffffffff90000000;
```
Comment (no92) [inline kernel/thor/arch/riscv/link.x:5]: Is this necessary? UEFI eir is identity mapped, so it shouldn't interfere with thor no matter what.

File: kernel/thor/arch/riscv/debug.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#include <arch/aarch64/mem_space.hpp>
```
Comment (marv7000) [inline kernel/thor/arch/riscv/debug.cpp:1]: Is that on purpose?

File: kernel/thor/arch/riscv/link.x (hunk 1 above)
Comment (avdgrinten) [reply kernel/thor/arch/riscv/link.x:5]: Good point. This is a residue from Electrode's PR which used a different `eir`. We should be able to change it back.

File: kernel/thor/arch/riscv/thor-internal/arch/stack.hpp [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <stdint.h>
+
+namespace thor {
+
+template <typename F>
+inline void walkThisStack(F) {
+	// TODO
```
Comment (no92) [inline kernel/thor/arch/riscv/thor-internal/arch/stack.hpp:9]: Should this use the `unimplementedOnRiscv` stub too?

Comment (avdgrinten) [reply kernel/thor/arch/riscv/thor-internal/arch/stack.hpp:9]: This is an optional debugging function that aarch64 also doesn't implement.

**Review by marv7000 — APPROVED** [review]
LGTM

File: kernel/thor/arch/riscv/link.x
[the remark this answers is not in the document]
Comment (avdgrinten) [reply kernel/thor/arch/riscv/link.x:1]: Good point, I'll add that when we start adding proper functionality.

**Pull request merged**